src/bin/str.rs

```diff
@@ -54,30 +71,56 @@ impl fmt::Display for Constant {
 impl std::str::FromStr for Constant {
     type Err = ();
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match i64::from_str(s) {
-            Ok(i) => Ok(Self::Num(i)),
-            Err(_) => {
-                let s = s.trim_matches('"');
-                Ok(Self::Str(s.as_bytes().into()))
-            }
+        if let Ok(i) = i64::from_str(s) {
+            Ok(Self::Num(i))
+        } else {
+            Ok(Self::Str(SmallString::from_str(s.trim_matches('"'))))
+        }
+    }
+}
+
+impl Constant {
+    fn to_smallstr(&self) -> Option<SmallString<[u8; 24]>> {
```

Comment:
why do we need both to_str and to_smallstr?

Reply:
Probably simpler to just have to_str